src/compilation/compiler.rs

```diff
@@ -555,6 +553,14 @@ impl Compiler {
         }
     }
 
+    /// Compiles a negation expression.
+    unsafe fn compile_negation(&mut self, operand: LLVMValueRef, source_type: &Type) -> LLVMValueRef {
+        match source_type {
+            Type::Int(_) => LLVMBuildNeg(self.builder, operand, cstr!("neg")),
```

Comment:
```suggestion
            Type::Int(IntType { signed: true, .. }) => LLVMBuildNeg(self.builder, operand, cstr!("neg")),
```

Love it. One small thing: if we're checking that the typer did its job correctly, I think we should check that the type is signed. What do you think?